High Capacity Dropships

4hands44 opened the pull request:
Remaps Alamo and Normandy to seat 60 marines.

## About The Pull Request
Changes DS interiors to seat 60 marines.

## Why It's Good For The Game
makes it so Dropships can seat 60 people. No more having to make dog piles on the floor because nobody can sit down. HRP and soul. Realisticly why would 30 odd marines pile on top of each other on the floor?

https://gyazo.com/ecfe8f0813a799eb92be7759daee3cda

## Changelog

:cl:
add: Alamo and normandy now use APC First aid stations and Fire Cabinets to reflect their nature as vehicles
add: Remapped alamo and normandy
add: Added "S.E.R.E. Locker" to Dropships. Required access level to open cabinet is same as Cockpit
add: Phone to Hangar Control room
(Contains Renamed Rappel harnesses (Due to lack of a real parachute asset), Medkit, MREs, 1 m39, and gun but not spare ammo) Mainly intended for RP reasons and should be treated like a Red Alert Locker.
add: Phone to Hangar Control room
qol: less need to make dog piles on the floor, or APC running people over/blocking them.


/:cl:

makes it so Dropships can seat 60 people. No more having to make dog piles on the floor because nobody can sit down. HRP and soul.

SirVerbsAlot commented:
so.....much soul.

kiVts commented:
Looks odd but on second thought actually makes sense and thats how it should be

AnCopper commented:
This'll make it more difficult for Xenos to board the Alamo since they have no cover inside of the ship.

FlamermanGrills commented:
The 4 one tile IFF lanes go brrrrrr

Misttyy commented:
Looks a bit funky, yeah. But this is probably the best setup. 

No longer needing to fight over seats, plus APC no longer soft griefing POs by destroying modules. +1 

Xeno cover is a non-issue. Any hijack will have the Queen on it and they, alongside any drone or hive lord can build up cover within the DS.

gohdatrice commented:
When people talk about cover they're not talking about post-crash hijack. It's more about if some xenos get trapped on the dropship during evac, or when a runner/lurker boards the dropship during the round (which does happen, these are not super rare occurrences). This is a much worse place to fight for sure, both in the sense that it screws over xenos but also it's less interesting/fun to fight on in general.
But that's arguably a minor issue compared to making it so people can actually sit in the dropship. But it is still a sacrifice nonetheless, even if a minor one.

Misttyy commented:
Minor sacrifice, usually they end up dead anyways.

SirVerbsAlot commented:
> When people talk about cover they're not talking about post-crash hijack. It's more about if some xenos get trapped on the dropship during evac, or when a runner/lurker boards the dropship during the round (which does happen, these are not super rare occurrences). This is a much worse place to fight for sure, both in the sense that it screws over xenos but also it's less interesting/fun to fight on in general. But that's arguably a minor issue compared to making it so people can actually sit in the dropship. But it is still a sacrifice nonetheless, even if a minor one.

Yeah but this also makes it so if you get boiler gas in there or a queen screech its FAR more effective, basically wiping out the entire DS becomes super easy if you get that in there.

gohdatrice commented:
> Yeah but this also makes it so if you get boiler gas in there or a queen screech its FAR more effective, basically wiping out the entire DS becomes super easy if you get that in there.

Does that sound more fun though? It's not really about whether it's a nerf or a buff, it's about whether it's more fun (to me anyway). When I say it's a sacrifice I mean it sounds less fun to me. I've had some pretty fun fights in the dropship but that won't really happen anymore, which is sad, but I also like that I can actually sit in the dropship instead of having to lay on the floor at the beginning of a round

PhantomEpicness commented:
> This'll make it more difficult for Xenos to board the Alamo since they have no cover inside of the ship.

skill issue + boo nerf

PhantomEpicness commented:
If you really want to add some cover, I'd suggest adding directional windows between the seating sections

Wintermote commented:
We have needed this for actual years

tempelers commented:
So much soul with this one. You could make it so the third module spot (one in the main area) is closer to the cockpit to make the machine gun nest is actually useful. Besides that, this PR is amazing and I have no other gripes.

4hands44 commented:
> So much soul with this one. You could make it so the third module spot (one in the main area) is closer to the cockpit to make the machine gun nest is actually useful. Besides that, this PR is amazing and I have no other gripes.

The issue with moving the modules further up is they block movement down the spine of the DS at full capacity, other wise I woulda kept them in the middle.

Mirvra commented:
This needs to get merged as soon as possible.

It just looks straight up beautiful compared to the horror we have right now.

Blithering commented:
Could you preserve the seat behind the foremost attach point? I know it's not strictly necessary but being able to sit in that while rear gunning or hoisting up medivacs is useful

GoldenDarkness55 commented:
rip zerker rav

stanalbatross reviewed (changes requested):
![image](https://user-images.githubusercontent.com/66756236/190855285-f7a9aa85-4e56-4908-acb7-d295be3015bd.png)
floating camera and intercom looks goofy

AnCopper commented:
> ![image](https://user-images.githubusercontent.com/66756236/190855285-f7a9aa85-4e56-4908-acb7-d295be3015bd.png) floating camera and intercom looks goofy

I think that it's supposed to be mounted on the platform, but the illusion doesn't really work.

4hands44 commented:
> ![image](https://user-images.githubusercontent.com/66756236/190855285-f7a9aa85-4e56-4908-acb7-d295be3015bd.png) floating camera and intercom looks goofy

Moved Camera and intercom, Fixed a missing button from the Normandy that wasn't mirrored to the Alamo

Tried and failed to fix the Broken Door control button in CIC Checkpoint

https://gyazo.com/fc22e5e7904736ad3cb8fc1ae7d11ec5

Firecharge123 commented:
Me want update, very good yes yes.

Pyrodes commented:
> rip zerker rav

bro this dropship design looks even better for eviscerate lol

Agameofscones reviewed (approved):
I'd like to see this TM'ed before it is merged, just to see how the xenos play around it during hijack. But the mapping is A-OK.

4hands44 commented:
> I'd like to see this TM'ed before it is merged, just to see how the xenos play around it during hijack. But the mapping is A-OK.

That was my thought as well (to TM not full merge it) Since it's a rather dramatic alteration from what we've had for literal years

usnpeepoo commented:
Rip drop ship corner combat

tempelers commented:
I'm having issues installing Fulton's to the equipment points inside, nothing I do seems to work. At least for the Alamo.

4hands44 commented:
> I'm having issues installing Fulton's to the equipment points inside, nothing I do seems to work. At least for the Alamo.

This has been fixed. However the fix has not been merged to the test yet.

4hands44 commented:
##Known issues

Alamo Attach points not working: Status = #Fixed
https://gyazo.com/1395516c46ef1a37c0f626acec34818b

Heads being cut off by chairs: Status = #Fixed
https://gyazo.com/78d2c12d1935904ec6d1dcca5ff759ed

Being unable to sit in some chairs: Status = Cannot be fixed, Is known issue with APC chair Code which also randomly bugs out.
IF somebody has a solution I'm all ears. it seems hit or miss on if it lets everyone sit or not. May be a script error server side, since in offline tests its worked fine?

Dropship cameras bugging out: Status = Fixed
https://gyazo.com/df466fcb672552b1428ad70957d174c3

Missing lockdown on normandy hatch: Fixed

##Other Stuff:

Added "S.E.R.E. Locker" to Dropships.
(Contains Renamed Rappel harnesses(due to lack of a real parachute asset), Medkit, MREs, 1 m39, and gun but not spare ammo) Mainly intended for RP reasons.
Required access level to open cabinet is same as Cockpit
https://gyazo.com/da056be7a8bf562687f740ceea9bd776

Added Invisible Vehicle barrier so APC and other vehicles cannot back up over the rear Cargo bay ledges. This was annoying and could impede movement in crew compartment.
https://gyazo.com/776fdf6de28ca39c5fedb86d990a3397

Added stairs under the doors to closer represent the side doors shown in the movie.
https://static.wikia.nocookie.net/avp/images/9/95/Smart_Ass.png/revision/latest/top-crop/width/360/height/360?cb=20150112212420

Chainsawicus commented:
I think the south hardpoints need to move up and be nested beside the barrier, as is the APC is so squished in that it clips through the back hatch of the dropship.

![image](https://user-images.githubusercontent.com/103772903/191735560-3e515dd5-4ae7-4275-8316-569868c3ff95.png)

morrowwolf commented:
https://cdn.discordapp.com/attachments/985905303247798304/1022611007601918044/unknown.png

Medical items spawning in the wall, should be an offset from the tile in front of it iirc

morrowwolf reviewed (changes requested):
https://cdn.discordapp.com/attachments/985905303247798304/1022611007601918044/unknown.png

Medical items spawning in the wall, nanomed should be offset from the tile in front of it iirc

Wintermote commented:
This is going to be one of the best changes to the game in years

Kenju22 commented:
So, I LOVE this change, it's really great and an amazing job as I have praised quite a bit in game and on the Discord. Just a small suggestion based on observations from a few rounds would be to move the rear center interior slot back to the middle of the DS itself. The reason being for ease of use when using Repelling and MedEvac, you want that to be somewhere easy to reach with a lot of open space around it.

Legit I have zero complaints and that is just a suggestion, in no way a criticism of any sort. I love the changes, all the space, the fire extinguisher, the new truck space, new panic lockdown button, everything is a 10/10 with this :)

Mirvra commented:
Xenos can't build hive structures in like 95% of the ship, even if slashed the seats block construction.
Only place you can reliably build stuff is in the cargo area in the rear. otherwise you have to wiggle around like crazy trying to find the sweet spot to build ONE thing in the middle tiles, and any tiles remotely close to the seats are off limits.

4hands44 commented:
> https://cdn.discordapp.com/attachments/985905303247798304/1022611007601918044/unknown.png
> 
> Medical items spawning in the wall, nanomed should be offset from the tile in front of it iirc

fixed

Mirvra commented:
Noticed this just now, you can see one of the seats from outside even with fully blocked vision.
![6b365705d7fe2dd447b794cd03f2332b](https://user-images.githubusercontent.com/104951893/192075269-1341ed37-dc47-433b-aa1b-d3593828c67e.png)

stanalbatross commented on maps/map_files/USS_Almayer/USS_Almayer.dmm:
```diff
@@ -6073,6 +6073,7 @@
 	},
 /area/almayer/command/lifeboat)
 "aqL" = (
+/obj/item/clothing/head/bandana,
```
?

stanalbatross reviewed (commented):
Alright, I have a few issues with this
- the double seats just look ugly. Tiles are usually somewhat distinct from each other but putting two seats on each tile just results in a blur of pixels
- marines clip into each other and into walls with the double seats
- cargo bay area seems not sufficiently thought out
- the tops of the seats cover items above them
- severe lack of soul
- taking and holding the dropship is now far harder for xenos

I do not think these downsides are worth it

Mollwu commented:
![Sem título](https://user-images.githubusercontent.com/113233526/192101096-d0f8f276-2b51-47eb-9118-606895ddb2ec.png)
 
 Well then, i love the ship designs by it's entirety, but this i a thing that have been pestering me; If there was a wall to make one sided windows between the seats to prevent the "Strange offsets"  and "Sprite Decaptation" would be gods work.

4hands44 commented:
> Alright, I have a few issues with this
> 
>     * the double seats just look ugly. Tiles are usually somewhat distinct from each other but putting two seats on each tile just results in a blur of pixels
> 
>     * marines clip into each other and into walls with the double seats
> 
>     * cargo bay area seems not sufficiently thought out
> 
>     * the tops of the seats cover items above them
> 
>     * severe lack of soul
> 
>     * taking and holding the dropship is now far harder for xenos
> 
> 
> I do not think these downsides are worth it



Double seats ugly: If somebody wants to make nicer sprites I'm all ears

Clipping into each-other is just how Double seats work, What walls?

cargo bay area seems not sufficiently thought out, Do you want to expand on this so I can see what you mean? It's currently the same size as before, but now without Chairs to crush. Driver can park just fine, and hard points are in a place where they don't block anybody's movement, while still being possible to use with APC in back. Although it is intended for APC to park on them. All of this being done without changing the Dropship itself, only the interior.

the tops of the seats cover items above them: Layering issue. Can be fixed if someone tells me how to change layers.

severe lack of soul: Compared to LRP piles of people on the floor?

taking and holding the dropship is now far harder for xenos: Can you elaborate on this? I haven't seen any issues with them holding it.

Mirvra commented:
The only issue with holding dropship as xeno I've noticed so far is that you can't build hive structures anywhere near the double seats, even if slashed, you can apparently make nests, but no hive structures.

Honestly, if marines have managed to push to dropship doors then xenos haven't been holding the area around the crash well enough if you ask me. It's silly how easily some strong walls and a hive cluster can prevent marines from pushing dropship IF there's enough xenos to keep the pressure off builders long enough.

As marine I honestly enjoy the double seat dropship more, it feels more soul. No marines laying on the ground, most of the ''cargo'' is actually in the rear with the APC instead of taking a seat. And you can converse with some blueberry on the opposite side of the ship without having to go around the center column.

I do agree that layering needs work though. But sadly I have no idea how any of that works so I can't offer advice.

AmoryBlaine commented:
The seats need to be vertical, the interior looks really bad with horizontal seats. The APC doesn't need half of the shuttle dedicated to it.

Agameofscones commented:
Unfortunately we don't have sprite sorting on a global Y axis (only per tile, overlap here is caused by offsets. Literally a byond moment) so the overlap issues are effectively unfixable. Vertical double seating is possible, and may look nicer. _However_ this would be difficult to communicate to players when someone is already seated, since they'll be covered by one of the chairs anyway.

morrowwolf commented:
This was a really cool idea but it seems to have issues that just can't be hammered out effectively with how it is right now. I hope what has been learned here can be brought back in a more refined form in the future.
